frankDoc/src/test/java/nl/nn/adapterframework/frankdoc/testtarget/examples/exclude/Child.java

```diff
@@ -0,0 +1,10 @@
+package nl.nn.adapterframework.frankdoc.testtarget.examples.exclude;
+
+// It is important that this class does not have shown attributes.
+// It only has excluded attributes. We want to test that GroupCreator
+// considers it properly.
+public class Child extends Parent {
+	/** @ff.noAttribute */
+	public void setChildExcludedAttribute(String value) {
+	}
```

Comment:
Ik denk dat wanneer je een methode overschrijft, deze weer als 'actief / non-suppressed' mag zien!

Reply:
Een method die je override-d wordt ook weer actief / niet-suppressed, tenminste als het geen technische override is. Dit wordt getest in package nl.nn.adapterframework.frankdoc.testtarget.examples.ignore.attributes.GrandChild. Die code is al gemerged dus die zie je niet in deze pull request.

Deze test gaat over iets anders. Class ...exclude.Child wordt gebruikt als parent van ...exclude.GrandChild door het algoritme dat de attribuut groepen opbouwt. Dit algoritme slaat parent classes over als ze geen attributes definiëren. Maar parents die alleen attributes weglaten mogen niet worden vergeten. Class ...exclude/Child heeft één attribute setter met een @ff.noAttribute JavaDoc tag. Hij definiëert dus alleen weg te laten attributes, maar moet worden meegenomen door het algoritme.

Reply:
Heb je het nu over het gebruik van de `@Override` annotatie?

Reply:
Het gaat er niet om of er een @Override annotatie staat of niet. Ik heb een bug gevonden waarbij attributes niet altijd goed worden weggelaten in reactie op een @ff.noAttribute annotatie. De bug zorgde ervoor dat derived class GrandChild niet zag dat hij "childExcludedAttribute" niet mocht erven van Child.

Reply:
En over class nl.nn.adapterframework.frankdoc.testtarget.examples.ignore.attributes.GrandChild kan ik ook het volgende nog zeggen. De class ziet er als volgt uit:

```
package nl.nn.adapterframework.frankdoc.testtarget.examples.ignore.attributes;

public class GrandChild extends SuppressingChild{
	// Not re-introduced, technical override
	@Override
	public void setSuppressedAttribute(String value) {
	}

	/** Documented, so re-introduced as attribute */
	@Override
	public void setSuppressedAttribute2(String value) {
	}
}
```

De tweede attribute setter heeft een JavaDoc comment maar de eerste niet. De eerste heeft ook verder geen Frank!Doc gerelateerde annotaties. Het is een kale override waar geen betekenis aan wordt toegevoegd. Dit noem ik een "technical override". Ik ga ervan uit dat deze override er alleen om technische redenen is, zonder dat de betekenis van de method verandert. Daarom wordt de eerste override genegeerd. Het attribute wordt niet opnieuw ingevoerd.

De tweede attribute setter heeft een JavaDoc comment. Hier verandert de betekenis. Het is geen technische override en dit attribute wordt wel toegevoegd aan GrandChild.